bsip-0022.md

```diff
@@ -37,34 +37,202 @@ Currently within Bitshares DPOS system, an user's votes for committee/witness/pr
+
+
+## <div id="account-voting-stake" /> Voting Stake of an Account
+
+The full power of an account's vote equals the [the sum of](https://github.com/bitshares/bitshares-core/blob/bf7ff54d9a17aa43f4663521e371b8c0ddfc2284/libraries/chain/db_maint.cpp#L1184-1186) an account's liquid core tokens (BTS), core tokens in vesting balances, and core tokens in open orders including [collateral that is used for shorting smart assets into existence](https://github.com/bitshares/bitshares-core/blob/bf7ff54d9a17aa43f4663521e371b8c0ddfc2284/libraries/chain/market_evaluator.cpp#L234-245).
+
+_The full power of an account's vote will automatically be updated to reflect any changes to the liquid balance, vesting balance, or core tokens in open orders_.
+
+This proposal shall retain the calculation for the maximum voting stake of an account.
+
+
+## <div id="vote-decay-function" /> Voting Power Decay Function
+
+![Qualitative decay of voting power](bsip-0022/general-vote-decay.png)
+
+The voting power of an [account's voting stake](#account-voting-stake) shall be set to full voting power whenever an account's vote is re-cast for _any_ [referendum category](#referendum-categories).  The voting power shall remain at 100% for each of the referendum categories for the entire ["full-power period"](#vote-decay-parameters).
+
+After the full-power period, the voting power shall begin to decay towards zero during a "decay period" for every decayable referendum category.  Decay shall occur in a stepped fashion (TODO: pending confirmation by Customminer) during the decay period.  The [quantity of decay steps (n) shall be determined by the BitShares Committee](#vote-decay-parameters).  There shall be (n - 1) decay periods.  The duration of each stepped voting period (sub-period) shall equal the total vote decay duration divided by the number of decay periods.  The voting power (P<sub>i</sub>) during decay period i shall be
+
+P<sub>i</sub> = (n - i) &div; n = 1 - (i &div; n)
+
+After the last decay period has expired, the voting power shall reduce to zero until either the account re-casts its votes, or the [vote decay parameters are adjusted](#vote-decay-parameters) such that the age of the account's last vote within that category possibly moves from "expired" to the the "decay age" or "full-power age".
+
+In summary, the age of a category vote by an account will typically decay from "full-power" through "decaying" to "expired" unless and until the account re-casts its voting slate in any referendum category.
+
+---
+<center>**Example of Vote Decay**</center>
+
+The vote decay parameters for witness voting have been set to 365 days for the full-power duration, 180 days for the total vote-decay duration, and 3 vote decay steps.  Each vote-decay sub-period has a duration of 60 days.
+
+A voter changes their vote for witnesses on December 31, 2019, and then does not vote again until December 31, 2025.
```

Comment:
I suppose he does vote on Jan 1, 2026?

Reply:
> I suppose he does vote on Jan 1, 2026?

The vote tallying mechanism has not yet been explicitly described in the BSIP.  If it occurs only during the current maintenance interval which is currently once every 24 hours, if I remember correctly, then a vote on December 31, 2025 might not be detected until the following day.  Should a sentence be added here that states the assumption that vote tallying will occur during the next maintenance interval which is assumed to occur on the following day?

Should the BSIP explicitly specify that vote decay will be done as part of the current vote tally activity?  Should this be left to the code technical specifications?

The frequency of vote tally is also related to your performance question.

Reply:
A few lines further down you say that he has 100% voting power starting Jan 1 2026, so I assume he must have voted at that point?

Yes, it might be a good idea to specify that vote decay takes effect during vote tallying, which happens in the maintenance interval.